### PR TITLE
cf 2set

### PR DESCRIPTION
**Pull request by envyaims**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by envyaims** [comment]
is it supposed to autorequest ben or did i screw something up

File: solutions/cf-468B.mdx
Comment (bqi343) [inline solutions/cf-468B.mdx:1]: wrong folder

Comment (envyaims) [reply solutions/cf-468B.mdx:1]: i used the "commit code" feature in live editor...

Comment (envyaims) [reply solutions/cf-468B.mdx:1]: https://usaco.guide/editor

Comment (envyaims) [reply solutions/cf-468B.mdx:1]: @danielzsh might wanna fix this yo

File: content/4_Gold/DSU.problems.json [hunk 1]
```diff
@@ -147,8 +147,7 @@
       "isStarred": false,
       "tags": ["DSU", "2SAT"],
```
Comment (SansPapyrus683) [inline content/4_Gold/DSU.problems.json:148]: can you remove the 2sat tag

File: solutions/gold/cf-468B.mdx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+struct DSU {
```
Comment (SansPapyrus683) [inline solutions/gold/cf-468B.mdx:26]: code snip + use the one from the module for consistency

File: solutions/gold/cf-468B.mdx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+		}
+		if (at.count(b - p[i])) {
+			can_B[i] = true;
+			dsu.unite(i, at[b - p[i]]);
+		}
+	}
+	/*
+	first bit activated if all numbers in component i can reside in
+	graph A, and similarily the second bit for graph B
+	*/
```
Comment (SansPapyrus683) [inline solutions/gold/cf-468B.mdx:70]:
```suggestion
	/*
	 * first bit activated if all numbers in component i can reside in
	 * graph A, and similarly the second bit for graph B
	*/

File: solutions/gold/cf-468B.mdx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+		b = get(b);
+		if (sz[a] < sz[b]) { swap(a, b); }
+		if (a != b) {
+			p[b] = a;
+			sz[a] += sz[b];
+		}
+	}
+};
+
+int main() {
```
Comment (SansPapyrus683) [inline solutions/gold/cf-468B.mdx:48]: space out your main jesus christ

File: solutions/gold/cf-468B.mdx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+	for (int i = 0; i < n; i++) {
+		int comp_mask = can_component[dsu.get(i)];
+		if (comp_mask == 1) {
+			// only can be in set A
+			cout << 0 << " ";
+		} else {
+			// can be in either set A or B
+			cout << 1 << " ";
+		}
+	}
```
Comment (SansPapyrus683) [inline solutions/gold/cf-468B.mdx:94]: end your output w/ a newline

File: solutions/gold/cf-468B.mdx [hunk 6]
```diff
@@ -0,0 +1,100 @@
+---
+id: cf-468B
+source: CF
+title: Two Sets
+author: Chongtian Ma
+---
+
```
Comment (SansPapyrus683) [inline solutions/gold/cf-468B.mdx:7]: any hints?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
how long has it been

File: solutions/gold/cf-468B.mdx (hunk 2 above)
Comment (envyaims) [reply solutions/gold/cf-468B.mdx:26]: do i have to

File: solutions/gold/cf-468B.mdx (hunk 6 above)
Comment (envyaims) [reply solutions/gold/cf-468B.mdx:7]: idk im not good at making hints

File: solutions/gold/cf-468B.mdx (hunk 5 above)
Comment (envyaims) [reply solutions/gold/cf-468B.mdx:94]: theres only 1 test case?

File: solutions/gold/cf-468B.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/gold/cf-468B.mdx:34]: you can always have ryan merge

File: solutions/gold/cf-468B.mdx (hunk 5 above)
Comment (SansPapyrus683) [reply solutions/gold/cf-468B.mdx:104]: it's always good to do so

File: content/4_Gold/DSU.problems.json [hunk 7]
```diff
@@ -98,6 +98,18 @@
       "solutionMetadata": {
         "kind": "internal"
       }
+    },
+    {
+      "uniqueId": "cf-468B",
+      "name": "Two Sets",
+      "url": "https://codeforces.com/contest/468/problem/B",
+      "source": "CF",
+      "difficulty": "Normal",
```
Comment (SansPapyrus683) [inline content/4_Gold/DSU.problems.json:107]: you underestimate your intelligence

Comment (envyaims) [reply content/4_Gold/DSU.problems.json:107]: naur

File: solutions/gold/cf-468B.mdx (hunk 5 above)
Comment (envyaims) [reply solutions/gold/cf-468B.mdx:104]: 💀

File: solutions/gold/cf-468B.mdx (hunk 2 above)
Comment (ryanchou-dev) [reply solutions/gold/cf-468B.mdx:34]: huh, i didn't realize that the implementation for the concept switched over to one with classes...

i feel like that can be pretty unreadable too :p

probably shouldn't use your own, but this was the past implementation 

```cpp
struct DSU {
	vector<int> e;
	DSU(int N) { e = vector<int>(N, -1); }

	// get representive component (uses path compression)
	int get(int x) { return e[x] < 0 ? x : e[x] = get(e[x]); }
	
	bool same_set(int a, int b) { return get(a) == get(b); }
	
	int size(int x) { return -e[get(x)]; }
	
	bool unite(int x, int y) {  // union by size
		x = get(x), y = get(y);
		if (x == y) return false;
		if (e[x] > e[y]) swap(x, y);
		e[x] += e[y]; e[y] = x;
		return true;
	}
};
```
if you'd like to use that one, im more than happy to merge :D

**Review by ryanchou-dev — COMMENTED** [review]
⚡

**Comment by envyaims** [comment]
alas i cant escape the compressed and obfuscated benq template

**Comment by envyaims** [comment]
i'm going bald before this gets merged

File: solutions/gold/cf-468B.mdx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+// BeginCodeSnip{DSU}
+struct DSU {
```
Comment (SansPapyrus683) [inline solutions/gold/cf-468B.mdx:38]: use the module's dsu and i'll merge ❤️

Comment (envyaims) [reply solutions/gold/cf-468B.mdx:38]: is this not benqs dsu?? certainly isnt mine

Comment (SansPapyrus683) [reply solutions/gold/cf-468B.mdx:38]: what does this have to do with not wanting to use the modulel's dsu

Comment (envyaims) [reply solutions/gold/cf-468B.mdx:38]: whats the module dsu?

Comment (SansPapyrus683) [reply solutions/gold/cf-468B.mdx:38]: https://usaco.guide/gold/dsu#implementation

Comment (envyaims) [reply solutions/gold/cf-468B.mdx:38]: wtf ok

Comment (envyaims) [reply solutions/gold/cf-468B.mdx]: im never making another dsu internal sol agian

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**